src/sentry/workflow_engine/processors/workflow.py

```diff
@@ -0,0 +1,46 @@
+from sentry.eventstore.models import GroupEvent
+from sentry.workflow_engine.models import Detector, DetectorWorkflow, Workflow
+from sentry.workflow_engine.types import DetectorType
+
+
+def get_detector_by_event(evt: GroupEvent) -> Detector:
+    issue_occurrence = evt.occurrence
+
+    if issue_occurrence is None:
+        # TODO - Don't hardcode the type as a string, make an enum
+        detector = Detector.objects.get(project_id=evt.project_id, type=DetectorType.ERROR)
+    else:
+        detector = Detector.objects.get(id=issue_occurrence.evidence_data.get("detector_id", None))
+
+    return detector
+
+
+def evaluate_workflow_triggers(workflows: set[Workflow], evt: GroupEvent) -> set[Workflow]:
+    triggered_workflows: set[Workflow] = set()
+
+    for workflow in workflows:
+        if workflow.evaluate_trigger_conditions(evt):
+            triggered_workflows.add(workflow)
+
+    return triggered_workflows
+
+
+def process_workflows(evt: GroupEvent):
```

Comment:
can we pass additional information alongside the event here in a dict? there is some information outside of the event that is calculated and passed along that would be evaluated in DataConditions

https://github.com/getsentry/sentry/blob/a53b18fab531059d1e94dfcdfdd5f9698f9d93a7/src/sentry/tasks/post_process.py#L997-L1002

but if we can add it to the GroupEvent itself, we wouldn't need to pass the extra info dict

Reply:
we can probably pass the job to here rather than the group_event if needed, but i think some of these also feel like replicated data from the group_event model itself. for example, how is `is_new` different from `group_event.group.times_seen == 1`.

I think the best case scenario is that we _don't_ pass the job though, i think it's generally better if we can understand what's happening from the group event model itself, since that's what the detector is creating in the end.

Reply:
Some of these are fairly complicated like `is_regression` is calculated [here](https://github.com/getsentry/sentry/blob/master/src/sentry/event_manager.py#L1638) and it'd be nice to not have to recalculate if it can be passed in.

Reply:
👍 there's a thread with snigda that discusses how we might be able to align on just using the group event: https://sentry.slack.com/archives/C07H2DHMSS0/p1733177862010779

worst case, this could be a fairly easy update to take in a `PostProcessJob` and just get the group event from that. I'm hoping to keep this as just the GroupEvent if possible to make this method a bit more reusable. (we can also add another processing method in post_process that invokes process_workflows).

i'm figuring we kick the can down the road a little on this in particular since the workflow_engine platform just needs the occurrence right now.